lib/sort_by_length.rb

```diff
@@ -1,7 +1,18 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(1)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  sentence_array = my_sentence.split(" ")
+  sentence_array.length.times do
+    i = 0
+    sentence_array.each do |string|
```

Comment:
The loop should start at index `i` instead of starting over at the beginning of the array each time.